frontend/task/index.ts

```diff
@@ -0,0 +1,559 @@
+                                return function () {
+                                    context.propagate(sc);
+                                    func.apply(sc, arguments);
+                                }
+                            }(subContexts[i], subContexts[i][prop]);
+                        } else {
+                            context[prop] = subContexts[i][prop];
+                        }
+                    }
+                }
+
+                return context;
+            };
+        },
+
+        register: function (name, func) {
+            if (this.order.indexOf(name) > -1) {
+                return;
+            }
+            this.libs[name] = func;
+            this.order.push(name);
+        }
+    };
+
+    // Initialize with contexts loaded before
+    if (window.quickAlgoLibrariesList) {
+        for (let i = 0; i < window.quickAlgoLibrariesList.length; i++) {
+            window.quickAlgoLibraries.register(window.quickAlgoLibrariesList[i][0], window.quickAlgoLibrariesList[i][1]);
+        }
+    }
+}
+
+export function extractLevelSpecific(item, level) {
+    if ((typeof item != "object")) {
+        return item;
+    }
+    if (Array.isArray(item)) {
+        return item.map((val) => {
+            return extractLevelSpecific(val, level);
+        });
+    }
+    if (item.shared === undefined) {
+        if (item[level] === undefined) {
+            let newItem = {};
+            for (let prop in item) {
+                newItem[prop] = extractLevelSpecific(item[prop], level);
+            }
+            return newItem;
+        }
+        return extractLevelSpecific(item[level], level);
+    }
+    if (Array.isArray(item.shared)) {
+        let newItem = [];
+        for (let iElem = 0; iElem < item.shared.length; iElem++) {
+            newItem.push(extractLevelSpecific(item.shared[iElem], level));
+        }
+        if (item[level] != undefined) {
+            if (!Array.isArray(item[level])) {
+                console.error("Incompatible types when merging shared and " + level);
+            }
+            for (let iElem = 0; iElem < item[level].length; iElem++) {
+                newItem.push(extractLevelSpecific(item[level][iElem], level));
+            }
+        }
+        return newItem;
+    }
+    if (typeof item.shared == "object") {
+        let newItem = {};
+        for (let prop in item.shared) {
+            newItem[prop] = extractLevelSpecific(item.shared[prop], level);
+        }
+        if (item[level] != undefined) {
+            if (typeof item[level] != "object") {
+                console.error("Incompatible types when merging shared and " + level);
+            }
+            for (let prop in item[level]) {
+                newItem[prop] = extractLevelSpecific(item[level][prop], level);
+            }
+        }
+        return newItem;
+    }
+    console.error("Invalid type for shared property");
+}
+
+export function getFixtureData () {
+    return {
```

Comment:
Move example data, even temporary, into another file, to be kept even when the actual `getFixtureData` will be implemented, as example data is useful for tests or other development purposes.